### PR TITLE
[tensorflow][build]  Reverting tf2.3 changes to only upgrade systemd

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### DLC image/dockerfile

tensorflow/training/docker/2.3/py3/Dockerfile.cpu
tensorflow/training/docker/2.3/py3/cu102/Dockerfile.gpu
tensorflow/training/docker/2.3/py3/cu110/Dockerfile.gpu

### Additional context

Reverting the `apt-get upgrade` changes to just upgrade systemd package.

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mseth10 — DISMISSED** [review]
LGTM

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -13,17 +13,17 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-skip_frameworks = []
+skip_frameworks = ["mxnet", "pytorch", "huggingface_tensorflow", "huggingface_pytorch"]
```
Comment (SergTogul) [inline dlc_developer_config.toml:16]: please add autogluon here as well

File: tensorflow/training/docker/2.3/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -35,7 +35,7 @@ ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/est
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=1.0.9
 
-RUN apt-get update && apt-get -y upgrade && apt-get install -y --no-install-recommends \
+RUN apt-get update && apt-get -y upgrade --only-upgrade systemd && apt-get install -y --no-install-recommends \
```
Comment (saimidu) [inline tensorflow/training/docker/2.3/py3/Dockerfile.cpu:38]: nit: Please break this into multiple lines:
```suggestion
RUN apt-get update \
 && apt-get -y upgrade --only-upgrade systemd \
 && apt-get install -y --no-install-recommends \
```

File: tensorflow/training/docker/2.3/py3/cu102/Dockerfile.gpu [hunk 3]
```diff
@@ -35,7 +35,7 @@ ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/est
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=1.0.9
 
-RUN apt-get update && apt-get -y upgrade && apt-get install -y --no-install-recommends --allow-unauthenticated \
+RUN apt-get update && apt-get -y upgrade --only-upgrade systemd && apt-get install -y --no-install-recommends --allow-unauthenticated \
```
Comment (saimidu) [inline tensorflow/training/docker/2.3/py3/cu102/Dockerfile.gpu:38]: nit: Please break this into multiple lines:
```suggestion
RUN apt-get update \
 && apt-get -y upgrade --only-upgrade systemd \
 && apt-get install -y --no-install-recommends --allow-unauthenticated \
```

File: tensorflow/training/docker/2.3/py3/cu110/Dockerfile.gpu [hunk 4]
```diff
@@ -36,7 +36,7 @@ ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/est
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=1.0.9
 
-RUN apt-get update && apt-get -y upgrade && apt-get install -y --no-install-recommends --allow-unauthenticated \
+RUN apt-get update && apt-get -y upgrade --only-upgrade systemd && apt-get install -y --no-install-recommends --allow-unauthenticated \
```
Comment (saimidu) [inline tensorflow/training/docker/2.3/py3/cu110/Dockerfile.gpu:39]: Please break this into multiple lines:
```suggestion
RUN apt-get update \
 && apt-get -y upgrade --only-upgrade systemd \
 && apt-get install -y --no-install-recommends --allow-unauthenticated \
```

**Review by saimidu — COMMENTED** [review]
You will need to force TF 2.3 to build by copying the contents of:
https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec-2-3.yml
into
https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec.yml

File: tensorflow/training/docker/2.3/py3/Dockerfile.cpu (hunk 2 above)
Comment (arjkesh) [reply tensorflow/training/docker/2.3/py3/Dockerfile.cpu:38]: Can you add this line on top of the systemd upgrade statement?

```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: tensorflow/training/docker/2.3/py3/cu102/Dockerfile.gpu (hunk 3 above)
Comment (arjkesh) [reply tensorflow/training/docker/2.3/py3/cu102/Dockerfile.gpu:38]: Can you add this line on top of the systemd upgrade statement?

```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: tensorflow/training/docker/2.3/py3/cu110/Dockerfile.gpu (hunk 4 above)
Comment (arjkesh) [reply tensorflow/training/docker/2.3/py3/cu110/Dockerfile.gpu:39]: Can you add this line on top of the systemd upgrade statement?

```
# TODO: Remove systemd upgrade once it is updated in base image
```

File: dlc_developer_config.toml (hunk 1 above)
Comment (saimidu) [reply dlc_developer_config.toml:16]: The jobs don't seem to have been started even though `autogluon` hasn't been included here.

**Pull request closed without merging**